### ISSUE TITLE
Questions about LLM2CLIP finetune with own dataset

### ISSUE THREAD

**Issue by MrPanda007** [microsoft/LLM2CLIP#27]
I used the llama-3.2-1B-CC-Finetuned model extract text features from Chinese data, and finetune EVA02 using this data with the default configuration parameters in the project.
However, the metrics of the finetuned model on the Chinese dataset are very low(worse than finetuned with English dataset CC3M). Could you please explain why this might be happening?

**Comment by Yif-Yang** [comment]
Let me clarify some details first. When you mentioned that the Chinese fine-tuning performs worse, what are you comparing it to? Are you comparing it to the 8B version in our paper, or to a 1B model fine-tuned on Chinese and English data simultaneously? I think there are a few possibilities:  

1. You might be using our CC fine-tuned LLaMA 1B model, which was fine-tuned on purely English data. If you include Chinese data in the LLM2CLIP fine-tuning process, the results might improve significantly.  
2. I suggest trying mixed Chinese-English training, which might yield better performance.

**Comment by MrPanda007** [comment]
@Yif-Yang The situation is, I have a Chinese image-text retrieval dataset:

1. Using your open LLM2CLIP-EVA02 model (LLM is LLama-3-8B-CC-Finetuned), without llm2clip fine-tuning, the metric on this dataset is 59.77.
2. Using your open LLM2CLIP-EVA02 model (LLM is LLama-3.2-1B-CC-Finetuned) as the base model, and fine-tuning LLM2CLIP with 1 million Chinese data samples, the metric of the epoch-1 model is only 34.95.
3. Using your open LLM2CLIP-EVA02 model (LLM is LLama-3.2-1B-CC-Finetuned) as the base model, and fine-tuning LLM2CLIP with 1 million Chinese data samples and 2 million CC3M data samples, the metric of the epoch-1 model is only 9.11.

**Comment by Yif-Yang** [comment]
I tend to think this might be a bug, like an issue with the code or optimization, as it doesn’t seem quite normal.  

1. Did you enable the IIM gradient?  
2. Is the LLama-3.2-1B-CC-Finetuned paired with the ViT you’re using? Is it your own ViT or the one provided with the model? Could you test its performance without fine-tuning to check if there’s an issue with the loading process?

**Comment by MrPanda007** [comment]
@Yif-Yang I will focus on these two points and reach out to you if I have any further questions.

**Comment by MrPanda007** [comment]
I retrained the model with the following configurations:
    1. Single GPU was used for local training.
    2. Both the LLM (LLama3-8B-CC-Finetune) and VIT (EVA02) are models provided LLM2CLIP officially.
 
The model metrics are as follows:
    1. The metric of the model without fine-tuning is 59.77.
    2.The metric of the model after fine-tuning is 32.8.

The configuration parameters for model training are:
MODEL=EVA02-CLIP-L-14-336
PRETRAINED=eva_clip
python -m torch.distributed.launch --nproc_per_node=1 \
	--use_env training/main.py \
        --enable-deepspeed \
        --grad-accumulation-steps=1 \
        --grad-checkpointing \
        --name="T_vitl336_Rcc12mR_Rcc3m_4ep" \
        --save-frequency 1  \
        --zeroshot-frequency 1 \
        --report-to="tensorboard" \
        --wandb-project-name="LLM2CLIP" \
        --wandb-notes="EVA02-CLIP-L-14-336" \
        --train-data-list "data/own_ch_dataset/train-{00..0150}.tar" \
        --train-num-samples-list 1050000  \
        --pretrained=${PRETRAINED} \
        --dataset-resampled \
        --precision "fp16" \
        --warmup 0 \
        --batch-size=1024 \
        --eval-batch-size=1024 \
        --log-every-n-steps 50 \
        --epochs=4 \
        --lr=1e-5 \
        --visual-lr=1e-5 \
        --text-lr=1e-5 \
        --wd=0.05 \
        --visual-wd=0.05 \
        --text-wd=0.05 \
        --ld=1.0 \
        --text-ld=1.01 \
        --visual-ld=0.85 \
        --grad-clip-norm=5.0 \
        --smoothing=0. \
        --workers=8 \
        --model=${MODEL} \
        --seed 4096 \
        --gather-with-grad \
        --local-loss \
        --force-custom-clip \
        --optimizer="ap_adamw" \
        --zero-stage=1 \
        --dataset-type "webdataset"

**Comment by wswaq** [comment]
I have a detailed question. The script you used above imported the original EVA-CLIP checkpoints. Did you import the checkpoints we provided in any other location?

**Comment by MrPanda007** [comment]
@wswaq Actually, the pretrained model configured in my script is    ./model_zoo/llm2clip/LLM2CLIP-EVA02-L-14-336/LLM2CLIP-EVA02-L-14-336.pt

**Comment by MrPanda007** [comment]
I am using the code below to extract and save text features. Is that correct?
`
from llm2vec import LLM2Vec
from transformers import AutoModel, AutoConfig, AutoTokenizer

# build the model
llm_model_path = './LLM2CLIP-Llama-3-8B-Instruct-CC-Finetuned'
config = AutoConfig.from_pretrained(llm_model_path)
model = AutoModel.from_pretrained(
    llm_model_path,
    config=config,
    trust_remote_code=True,
    torch_dtype=torch.float16
)
model.config._name_or_path = 'meta-llama/Meta-Llama-3-8B-Instruct'

tokenizer = AutoTokenizer.from_pretrained(llm_model_path)
tokenizer.pad_token = tokenizer.eos_token
tokenizer.padding_side = 'left'
l2v = LLM2Vec(
    model,
    tokenizer,
    pooling_mode='mean',
    max_length=512,
    skip_instruction=True
)

text_features = l2v.encode(text, convert_to_tensor=True)
text_features = torch.nn.functional.normalize(text_features, p=2, dim=1)
`

**Comment by MrPanda007** [comment]
I am using the code below to extract and save text features. Is that correct?

```
from llm2vec import LLM2Vec
from transformers import AutoModel, AutoConfig, AutoTokenizer

# build the model
llm_model_path = './LLM2CLIP-Llama-3-8B-Instruct-CC-Finetuned'
config = AutoConfig.from_pretrained(llm_model_path)
model = AutoModel.from_pretrained(
    llm_model_path,
    config=config,
    trust_remote_code=True,
    torch_dtype=torch.float16
)
model.config._name_or_path = 'meta-llama/Meta-Llama-3-8B-Instruct'

tokenizer = AutoTokenizer.from_pretrained(llm_model_path)
tokenizer.pad_token = tokenizer.eos_token
tokenizer.padding_side = 'left'
    l2v = LLM2Vec(
    model,
    tokenizer,
    pooling_mode='mean',
    max_length=512,
    skip_instruction=True
)

text_features = l2v.encode(text, convert_to_tensor=True)
text_features = torch.nn.functional.normalize(text_features, p=2, dim=1)
```

**Comment by MrPanda007** [comment]
@Yif-Yang Could you please help check the issue?

**Comment by raytrun** [comment]
The training data processing script has been uploaded. Please refer to [this](https://github.com/microsoft/LLM2CLIP/tree/main/llm2clip/data)

**Comment by 22wei22** [comment]
> [@Yif-Yang](https://github.com/Yif-Yang) Could you please help check the issue?

do you have new good result？

**Comment by MrPanda007** [comment]
@22wei22 Have done with the latest code.

**Comment by ctgushiwei** [comment]
@MrPanda007  hello,how to create our own dataset like this?

![Image](https://github.com/user-attachments/assets/6d02c812-40a8-4ed0-880d-be0fdceb87d2)
